### ISSUE TITLE
Can't get it working

### ISSUE THREAD

**Issue by RocketEnvy** [cosminadrianpopescu/vim-sql-workbench#1]
Here is what I have done so far:
1. I copied your files into my .vim directory
~$ ls ~/.vim
autoload  plugin  resources  syntax
1. I installed workbench in ~/workbench
2. I added this to my .vimrc: let g:sw_exe ="/home/rocket/workbench/sqlwbconsole.sh"
3. In the workbench gui I created a profile named Manifesto and verified that I can connect to the database.
4. In vim I tried: :SWSqlOpen Manifesto test ... which gives me a regular vim window, no database panes or anything unusual.
5. In vim I tried: :SWDbExplore Manifesto ... which vim responds with "E464: Ambiguous use of user-defined command"

I speak English natively so if you can tell me where I went wrong, I may be able to help clear up the relevant part of the documentation. 
:wq

**Comment by cosminadrianpopescu** [comment]
It's `SWDbExplorer`. Please note the missing r. Please let me know if like this is working. I will search the documentation and fix it if needed.

**Comment by cosminadrianpopescu** [comment]
I've fixed the documentation. I will close the issue. Please reopen it if it's not working with `SWDbExplorer`. Also, please note that for `SWSqlOpen`, this opens a normal VIM buffer. Then, you can go in normal mode with the cursor over a sql and click `ctrl + space`. This will execute the sql and return the result set. Also, you can customize your shortcut (see the documentation)

**Comment by sluedtke** [comment]
Sorry, I have as well problems to get it working. Do you want me to open another issue or post it at this place??

**Comment by cosminadrianpopescu** [comment]
What is the issue? Because now, SWDbExplorer command takes a new argument. Could you please let me know what command you tried and what was the result? Also, have you started the server? Please see the documentation on github. I haven't had time yet to change the vim documentation.

**Comment by sluedtke** [comment]
Hi,

I started in a fresh vim session:

:SWServerStart 5000
:SWSqlConnectToServer 5000
both work or at least do not give an error

:SWDbExplorer 5000 crop_db
that one fails with:

Error detected while processing function
sw#dbexplorer#show_panel..sw#server#open_dbexplorer..<SNR>131_pipe_execute:
line   10:
E605: Exception not caught: There is no port set for this buffer.

crop_db is a configured profile. invoking

sqlwbconsole.sh -profile= crop_db

from a shell works.

I am on ubunut 12.04, vim 7.4.5 and
let g:sw_exe="~/.local/bin/sqlwbconsole.sh" is in my vimrc

thanks for the effort.

Stefan

On 04/14/2015 12:40 PM, cosminadrianpopescu wrote:

> What is the issue? Because now, SWDbExplorer command takes a new argument. Could you please let me know what command you tried and what was the result? Also, have you started the server? Please see the documentation on github. I haven't had time yet to change the vim documentation. 
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-92763355

## 

Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by cosminadrianpopescu** [comment]
Try running the server from command line with `-d 1` (instead of running `SWServerStart` (see the server help if you need) like this: 

```
cd ~/.vim/bundle/vim-sql-workbench/resources
./sqlwbserver -t /tmp -c ~/.local/bin/sqlwbconsole.sh -o 5000 -d 1
```

and let me know what this outputs. Also, make sure that vim is compiled with and started in servermode. From vim do `echo v:servername`.

**Comment by sluedtke** [comment]
On 04/14/2015 01:01 PM, cosminadrianpopescu wrote:

> Try running the server from command line with `-d 1` (instead of
> running `SWServerStart` (see the server help if you need) like
> this:
> 
> `cd ~/.vim/bundle/vim-sql-workbench/resources ./sqlwbserver -t
> /tmp -c ~/.local/bin/sqlwbconsole.sh -o 5000 -d 1`
> this gives:

13:13 $ ./sqlwbserver -t /tmp -c ~/.local/bin/sqlwbconsole.sh -o 5000 -d 1
OPENING: /home/sluedtke/.local/bin/sqlwbconsole.sh -feedback=true
-showProgress=false
-e

SQL Workbench/J (117) console interface started.
Enter exit to quit.
Enter WbHelp for a list of SQL Workbench/J specific commands
Config directory: /home/sluedtke/.sqlworkbench

but  I do not get a cursor, only ctrl+ c works

> and let me know what this outputs. Also, make sure that vim is
> compiled
> it is compiled with the client server mode
> with and started in servermode. From vim do `echo v:servername`.
> the output is always "GVIM2" , no matter whether I start it with a
> without the `-servername` flag or not
> 
> --- Reply to this email directly or view it on GitHub: 
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-92769474
> 
> ## 
> 
> Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by cosminadrianpopescu** [comment]
It's normal to only have `ctrl+c`. Leave the server on (don't click `ctrl+c`) and then do again `SWSqlConnectToServer`. Then execute a statement. You should see some output in the server window. Please tell me what happens in the server window .

**Comment by sluedtke** [comment]
Nothing happens, no matter whether I start vim with or without on
explicit server name.

When I did ctrl+c, the shell does not work any more, commands are not
displayed but executed.

On 04/14/2015 01:30 PM, cosminadrianpopescu wrote:

> It's normal to only have `ctrl+c`. Leave the server on (don't click `ctrl+c`) and then do again `SWSqlConnectToServer`. Then execute a statement. You should see some output in the server window. Please tell me what happens in the server window .
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-92776284

## 

Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by cosminadrianpopescu** [comment]
Yes, this is a bug on the server (the shell not working anymore). Open a new shell, start the server, start the vim with the server mode enabled (make sure `echo v:servername` produces an output), then do `SWSqlConnectToServer 5000` on a buffer. In that buffer then type on a row `wbconninfo;` and then go with the cursor on top of the `wbconninfo` word and click `ctrl+space` and let me know what happens.

**Comment by sluedtke** [comment]
Again, nothing happens.

13:47 $ gvim –servername SAMPLESERVER test.sql
3 files to edit

I wonder why vim says "3 files to edit".

If I skip through the buffers (1-3) nothing is in there, apart from the
line I entered. Coming back to buffer 1, an error is given:

  ~
Error detected while processing function
sw#sqlwindow#check_results..sw#server#fetch_result..<SNR>125_pipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')

On 04/14/2015 01:44 PM, cosminadrianpopescu wrote:

> Yes, this is a bug on the server (the shell not working anymore). Open a new shell, start the server, start the vim with the server mode enabled (make sure `echo v:servername` produces an output), then do `SWSqlConnectToServer 5000` on a buffer. In that buffer then type on a row `wbconninfo;` and then go with the cursor on top of the `wbconninfo` word and click `ctrl+space` and let me know what happens. 
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-92780051

## 

Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by cosminadrianpopescu** [comment]
This means that your server is not started. See the error (`name socket is not defined`). This means that the server is not started. Could you please make sure that you have the server started working on the port 5000?

**Comment by sluedtke** [comment]
14:17 $ sudo netstat -atnp | grep 5000
13:tcp        0      0 127.0.0.1:5000          0.0.0.0:*
LISTEN      20131/python

If I do not start the server, grep does not return anything. Is that
right? Does this say the server started ?

I just figured out that only "vim" can be started with the
"--servername" argument.  Following your previous instructions, I get
the same error.

Do I have to set specific rights on sqlwbconsole.sh, apart from being
executable for myself.

Sorry, I did not expect that it will be that hard.

On 04/14/2015 02:06 PM, cosminadrianpopescu wrote:

> This means that your server is not opened. See the error (`name socket is not defined`). This means that the server is not started. Could you please make sure that you have the server started working on the port 5000? 
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-92784130

## 

Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by cosminadrianpopescu** [comment]
No, you don't have to have any specific rights.

Let's start from the beginning. Restart your computer .Then in a terminal do 
`sqlwbconsole.sh -profile= crop_db`

There do a `wbconninfo;` and press `enter`. This should output information about your current connection. If it doesn't, then let me know (it means you need to do a config change in sqlwbconsole). If it does produce the output, type `exit` and enter. This will close your `sqlwbconsole` connection. Then go again and start your `sqlwbserver`: 

```
cd ~/.vim/bundle/vim-sql-workbench/resources
./sqlwbserver -t /tmp -c ~/.local/bin/sqlwbconsole.sh -o 5000 -d 1
```

Then open a new terminal and test this connection. Do 
`telnet localhost 5000` and then type `DBEcrop_db` and then type `enter`. This should return on your screen `DISCONNECT`. If it's not the case, let me know. 

If you get this output, then it means your server is working. 

Then type `ctrl+]` and then `quit` and `enter`. This will take you out of `telnet`. 

If you got this far, open your `vim` and type ':SWDbExplorer crop_db 5000`. This should open a database explorer tab. If this is the case, you have everything working. It means that there is a problem with your sql buffer. 

Please let me know how this works.

**Comment by sluedtke** [comment]
On 04/14/2015 02:36 PM, cosminadrianpopescu wrote:

> No, you don't have to have any specific rights.
> 
> Let's start from the beginning. Restart your computer .Then in a
> terminal do `sqlwbconsole.sh -profile= crop_db`
> 
> There do a `wbconninfo;` and press `enter`. This should output
> information about your current connection. If it doesn't, then let
> me know (it means you need to do a config change in sqlwbconsole).
> If it does produce the output, type `exit` and enter. This will
> close your `sqlwbconsole` connection. Then go again and start your
> `sqlwbserver`:
> 
> `cd ~/.vim/bundle/vim-sql-workbench/resources ./sqlwbserver -t
> /tmp -c ~/.local/bin/sqlwbconsole.sh -o 5000 -d 1`
> 
> Then open a new terminal and test this connection. Do `telnet
> localhost 5000` and then type `DBEcrop_db` and then type `enter`.
> This should return on your screen `DISCONNECT`. If it's not the
> case, let me know.
> 
> If you get this output, then it means your server is working.

This works, but only once. If I try to do it again (starting with
telnet ...  after I successfully quited) I do not get a DISCONNECT.

> Then type `ctrl+]` and then `quit` and `enter`. This will take you
> out of `telnet`.
> 
> If you got this far, open your `vim` and type ':SWDbExplorer
> crop_db 5000`. This should open a database explorer tab. If this is
> the case, you have everything working. It means that there is a
> problem with your sql buffer.

This works in vim and gvim, the new tab has 3 screens (as it should
have I guess) but the shortcuts (O) (T) do not work. I still get the
same errors.

**Comment by cosminadrianpopescu** [comment]
> This works, but only once. If I try to do it again (starting with
> telnet ...  after I successfully quited) I do not get a DISCONNECT.

That's normal. 

> This works in vim and gvim, the new tab has 3 screens (as it should
> have I guess) but the shortcuts (O) (T) do not work. I still get the
> same errors.

Could you please give me again the error?

**Comment by sluedtke** [comment]
On 04/14/2015 03:08 PM, cosminadrianpopescu wrote:

> > This works, but only once. If I try to do it again (starting with
> > telnet ...  after I successfully quited) I do not get a DISCONNECT.
> 
> That's normal. 
> 
> > This works in vim and gvim, the new tab has 3 screens (as it should
> > have I guess) but the shortcuts (O) (T) do not work. I still get the
> > same errors.
> 
> Could you please give me again the error? 
> That is the output of ":messages" after SWDbExplorer

Messages maintainer: Bram Moolenaar Bram@vim.org
Error detected while processing function
sw#dbexplorer#show_panel..sw#server#open_dbexplorer..<SNR>123_pipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')
Error detected while processing function
sw#dbexplorer#show_panel..<SNR>122_set_highlights:
line    2:
E419: FG color unknown
Error detected while processing function
sw#dbexplorer#show_panel..<SNR>122_change_tab..sw#server#dbexplorer..<SNR>123_pipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')

And that after "O"

Error detected while processing function
<SNR>122_change_tab..sw#server#dbexplorer..<SNR>123_pipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')

**Comment by cosminadrianpopescu** [comment]
So, you did a `telnet localhost 5000` and then a `DBEcrop_dp` and this returned a `DISCONNECT`? Because if you did this, then the server is up and running properly. Also, did you do the `swsqlconsole.sh -p crop_db` and then the `wbconninfo;`. If this also worked, it means that you are somehow closing the server afterwards. You don't click and `ctrl+c` in the server terminal after doing the tests, are you?

**Comment by sluedtke** [comment]
I tried again, output follows below. And I took care that I did not
click or ctrl+c in the server window

✔ /tmp 
16:21 $ sqlwbconsole.sh -profile= crop_db
-e 

SQL Workbench/J (117) console interface started.
Enter exit to quit.
Enter WbHelp for a list of SQL Workbench/J specific commands
Config directory: /home/sluedtke/.sqlworkbench

Connection to "User=sluedtke, Schema=public, URL=jdbc:postgresql://139.17.99.27:5432/crop_generator" successful

sluedtke@crop_generator/public> wbconninfo
..> ;

  Product Name: PostgreSQL
  Product Version: 9.3
  Product Info: 9.3.6
  Driver Name: PostgreSQL Native Driver
  Driver Class: org.postgresql.Driver
  Driver Version: PostgreSQL 9.1 JDBC3 (build 901)
  URL: jdbc:postgresql://139.17.99.27:5432/crop_generator
  Isolation Level: READ COMMITTED
  Username: sluedtke
  Schema: public
  Database (Catalog): crop_generator
  Workbench DBID: postgresql
  Connection ID: Console

sluedtke@crop_generator/public> exit
-e 
✔ /tmp 
16:22 $ telnet localhost 5000
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
DBEcrop_db
DISCONNECT^]
telnet> quit
Connection closed.
✔ /tmp 
16:23 $

**Comment by cosminadrianpopescu** [comment]
Then could you please open vim and do `SWSqlConnectToServer 5000 /tmp/sql.sql`. 

Then in the newly created buffer type `wbconninfo;` and then click `ctrl+space`. 

Make sure you have a server window started and do `telnet localhost 5000` and then `DBEcrop_db` before and that you get a `DISCONNECT` reply. 

If the telnet test works and vim doesn't work, this is very strange. Please let me know of the vim reply. 

Thank you.

**Comment by sluedtke** [comment]
Hi,

so, it still does not work, not within vim!

The error is:

Error detected while processing function sw#sqlwindow#execute_sql..sw#variables#extract:
line    5:
E55: Unmatched )
E55: Unmatched )
Processing a command. Please wait...
Error detected while processing function sw#sqlwindow#execute_sql..sw#execute_sql..sw#server#execute
_sql..<SNR>126_pipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')

The error comes only with vim. In gvim, it fails silently and messages
shows nothing but ctrl+space shows nothing as well.

**Comment by cosminadrianpopescu** [comment]
I think that the problem is with the shortcuts and gvim. Could you try instead of doing `ctrl+space` to run the command `:SWSqlExecuteCurrent`?

**Comment by sluedtke** [comment]
I tried that and the error is:

Error detected while processing function sw#sqlwindow#extract_current_sql:
line   18:
E121: Undefined variable: b:delimiter
E116: Invalid arguments for function sw#sql_split(s, b:delimiter)
E15: Invalid expression: sw#sql_split(s, b:delimiter)
line   19:
E121: Undefined variable: sqls
E15: Invalid expression: sqls
line   27:
E605: Exception not caught: Could not identifiy the current query

I did start the server and tested via telnet..

**Comment by cosminadrianpopescu** [comment]
There is something strange. Can you do in vim `:let b:delimiter = ';'` and then try again `:SWSqlExecuteCurrent`?

**Comment by sluedtke** [comment]
If I do:

`:echo b:delimiter`, I get an undefined variable.

After `:SWSqlConnectToServer 5000 /tmp/test.sql` , the first statement
echos ";"

After setting it explicitly as your recommend, the
`:SWSqlExecuteCurrent` throws again the error:
Error detected while processing function
sw#sqlwindow#execute_sql..sw#variables#extract: line    5: E55:
Unmatched ) E55: Unmatched ) Processing a command. Please wait...
Error detected while processing function
sw#sqlwindow#execute_sql..sw#execute_sql..sw#server#execute_sql..<SNR>132_p
ipe_execute: line   53: Traceback (most recent call last):   File
"<string>", line 7, in <module> NameError: name 'socket' is not defined
line   54: E121: Undefined variable: result E116: Invalid arguments for
function len(result) <= 3 E15: Invalid expression: len(result) <= 3 line
  57: E121: Undefined variable: result E116: Invalid arguments for
function substitute(result, '\r', '', 'g') E15: Invalid expression:
substitute(result, '\r', '', 'g')

I looks like I forgot to call  `:SWSqlConnectToServer 5000
/tmp/test.sql` during my previous try, but I am pretty sure I have done
this. So b:delimiter is only  set after calling `SWSqlConnnectToServer`.

But still no luck!

**Comment by cosminadrianpopescu** [comment]
I don't know how to attach a file to this work item. I would like you to test another variables.vim file. 
I can send you the file on a private e-mail address. and let you know what to do with it or I could tell you how to  modify it. Please let me know how you prefer.

**Comment by sluedtke** [comment]
me neither ;-)

my mail is sluedtke@gfz-potsdam.de

On 04/15/2015 11:18 AM, cosminadrianpopescu wrote:

> I don't know how to attach a file to this work item. I would like you to test another variables.vim file. 
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/cosminadrianpopescu/vim-sql-workbench/issues/1#issuecomment-93279121

## 

Stefan Lüdtke

Section 5.4-  Hydrology
Tel.: +49 331 288 2821
Fax: +49 331 288 1570
Email: sluedtke@gfz-potsdam.de

Helmholtz-Zentrum Potsdam
Deutsches GeoForschungsZentrum GFZ
(GFZ German Research Centre for Geoscience)
Stiftung des öff. Rechts Land Brandenburg

## Telegrafenberg, 14473 Potsdam

PGP Public Key: http://bit.ly/13d9Sca

**Comment by sluedtke** [comment]
The error stays, but the unmatched  ... part is gone.

```
Processing a command. Please wait...
Error detected while processing function sw#sqlwindow#execute_sql..sw#execute_sql..sw#server#execute_sql..<SNR>132_p
ipe_execute:
line   53:
Traceback (most recent call last):
  File "<string>", line 7, in <module>
NameError: name 'socket' is not defined
line   54:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   57:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')    
```

**Comment by cosminadrianpopescu** [comment]
Somehow your port is not opened. Could you also try to do in vim `:echo b:port` in your sql buffer?

**Comment by sluedtke** [comment]
That gives "5000".

**Comment by cosminadrianpopescu** [comment]
Sorry, finally I got it. I forget to import the `socket` in vim. I don't know how it works in my case. Anyway, I am sending you by e-mail another file to test. It's the `.vim/bundle/vim-sql-workbench/autoload/sw/server.vim` file. Please let me know if now it works. 

Thank you.

**Comment by sluedtke** [comment]
Ok, that works now. At least I think so, because no error is thrown. 
In vim, the short cut works, in gvim it does not. 
Calling `:SWSqlExecuteCurrent` works for both. 

The output in the split buffer is:

```
  SQL> wbconninfo;                                                                                               
  No connection
  ^[]0;SQL Workbench/J^G
```

I will try other commands asap.

**Comment by cosminadrianpopescu** [comment]
You also need to do a `wbconnect crop_db;` in your buffer and execute that command (try with `ctrl+space`). But now it works. 

Please confirm that everything is ok and I will make a new release with the patch.

**Comment by sluedtke** [comment]
Back again :-)
The workflow we tested worked so far, but it does not, if the server is started from vim. 

```
:SWServerStart 5000
:SWSqlConnectToServer 5000 /tmp/test.sql
:SWSqlExecuteCurrent
```

The last one does not work and gives the following error:

```
Processing a command. Please wait...
Error detected while processing function sw#sqlwindow#execute_sql..sw#execute_sql..sw#server#execute
_sql..<SNR>132_pipe_execute:
line   54:
Traceback (most recent call last):
  File "<string>", line 9, in <module>
  File "/usr/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 111] Connection refused
line   55:
E121: Undefined variable: result
E116: Invalid arguments for function len(result) <= 3
E15: Invalid expression: len(result) <= 3
line   58:
E121: Undefined variable: result
E116: Invalid arguments for function substitute(result, '\r', '', 'g')
E15: Invalid expression: substitute(result, '\r', '', 'g')      
```

The command from the sql buffer I try to execute is:

`wbconnect crop_db;`

**Comment by cosminadrianpopescu** [comment]
Please note that the port here is 500 and then the port for the `SWSqlConnectToServer` is 5000. Is this a typo?

If not and it does not work, I will open another ticket with this, because I didn't really tested that much starting the server from `vim`. 

But, in the meanwhile, could you please confirm that if you start the server manually (from a terminal), everything works fine?

Thank you.

**Comment by sluedtke** [comment]
Sorry, that was a typo, I changed the entry above. 

Starting the server from (g)vim does not work. 

Not sure whether everything works or not. I just started some test. 

What does work!
- the procedure we used so far (starting the server manual y) and sending commands    with `:SWSqlExecuteCurrent` in vim and gvim
- the short cut (ctrl+space) in vim

What does not work:
- short cut in gvim  (ctrl +  space)
- the output of a query is displayed in the server terminal window
- the command `:SWDbExplorer` kills both, vim and gvim until  the terminal session  is killed.  After that, the explorer buffer stays empty but there was some output produced in the server terminal window.

So I can connect a vim buffer to the server, but I cannot reproduce those nice screenshots, but that might be another issue.

**Comment by cosminadrianpopescu** [comment]
So, the shortcut in vim works or not? 
Do you have the output of a query also displayed in vim? 
For the SWDbExplorer, you have to wait a little when you first open it (like 5, 10 seconds) in which vim will be frozen. This is because a new connection is spawned and vim waits for it before continuing. Can let it for a few seconds before killing it? And see what happens in the server window? 

Regarding the start of the server from vim, I opened the incident #8 
Until this incident is closed, please start the server manually from a terminal session.

**Comment by sluedtke** [comment]
The short cut woks only in vim.

No output  of the query displayed in vim nor gvim. 
Only the output of `wbconninfo` is shown in a split buffer.  But this is not always the case, sometimes it does, sometimes not.

I waited more than a minute after calling  SWDbExplorer. I get some output in the  server window, but only a part of the DB props. 

I will put the error message I get after starting the server from vim to #8.

**Comment by cosminadrianpopescu** [comment]
> The short cut woks only in vim.

I will fix the shortcut to work on vim and gvim (I only use vim, is true, so I didn't realize that is not working on gvim). Anyway, if you read the documentation, you can customize the shortcuts yourself. 

> No output of the query displayed in vim nor gvim.
> Only the output of wbconninfo is shown in a split buffer. But this is not always the case, sometimes it does, sometimes not.

Do you always get an output in the server window but that is not always sent in vim? Or when you don't get an output in the vim, you also don't get an output in the server window? 

> I waited more than a minute after calling SWDbExplorer. I get some output in the server window, but only a part of the DB props. 

What do you mean part of the DB props? I don't understand

**Comment by sluedtke** [comment]
I get at least output in the server window, so there is always output. 
In g(vim), I could not find a pattern yet. Sometime it splits, sometimes not. 

![screenshot1](https://cloud.githubusercontent.com/assets/5367675/7176953/3e21bd2c-e423-11e4-95b5-3657698f0097.png)

For the attached screenshot, the 2nd and 3rd statement are displayed in the server window only.  

Regarding the DBexplorer: I get a list of the public scheme in the server window (seq, tables, views ..) but nothing more. In (g)vim, a new tab is opened but nothing is in there. As soon as kill the server, (g)vim is working again and the top most panel of the DBexplorer tab has some content. `The current profile is crop_db`...

Other schemes are missing. They are shown in the sql-workbench gui.

**Comment by cosminadrianpopescu** [comment]
I got it. Apparently you have postgresql. I've only tested with mysql and oracle. I will do tests at home with postgressql and fix the issue. I will close this one and open another issue with postgresql not working.

See issue #9 

Thank you very much for your help. I hope that I will fix it and you will have it working in a day or two.

**Comment by sluedtke** [comment]
I thank your for your efforts!

**Comment by cosminadrianpopescu** [comment]
Once is working it will be worth it, I can assure you. If you want to use DBMS from vim, this is so much better than `dbext`. Not that `dbext` is bad. I've been using it before. But this is the next level. Myself, I've tested this plugin on `oracle`, `mysql`, `sql server` and `sqlite` and it worked perfectly.
